auth-api/src/auth_api/services/org.py

```diff
@@ -227,65 +228,84 @@ def _create_payment_settings(
         **kwargs,
     ):
         """Add payment settings for the org."""
-        pay_url = current_app.config.get("PAY_API_URL")
-        org_name_for_pay = f"{org_model.name}-{org_model.branch_name}" if org_model.branch_name else org_model.name
-        pay_request = {
-            "accountId": org_model.id,
-            # pay needs the most unique idenitfier.So combine name and branch name
-            "accountName": org_name_for_pay,
-            "branchName": org_model.branch_name or "",
-        }
+        try:
+            pay_url = current_app.config.get("PAY_API_URL")
+            org_name_for_pay = f"{org_model.name}-{org_model.branch_name}" if org_model.branch_name else org_model.name
+            pay_request = {
+                "accountId": org_model.id,
+                # pay needs the most unique idenitfier.So combine name and branch name
+                "accountName": org_name_for_pay,
+                "branchName": org_model.branch_name or "",
+            }
+            error_code = None
 
-        if payment_method:
-            pay_request["paymentInfo"] = {"methodOfPayment": payment_method}
+            if payment_method:
+                pay_request["paymentInfo"] = {"methodOfPayment": payment_method}
```

Comment:
Building the request should be refactored to a separate function. There are already alot of conditional branches that cause code complexity to go up

Reply:
If org service is getting too packed, consider building a pay service for this purpose.